ekncontent/ekncontent/eknc-domain.c

```diff
@@ -830,8 +858,13 @@ on_xapian_query_response (GObject *source,
           return;
         }
 
+      /* Note that we pass another closure here instead of just the task
```

Comment:
We would still need to share some data with query state though - in particular, the number of models which have been fetched asynchronously.